lib/datasource.js

```diff
@@ -2699,6 +2704,22 @@ DataSource.prototype.beginTransaction = function(options) {
   return Transaction.begin(this.connector, options);
 };
 
+/**
+ * Get the maximum number of event listeners
+ */
+DataSource.prototype.getMaxOfflineRequests = function() {
+  // Set max listeners to a default value
+  // Override this default value with a datasource setting
+  // 'maxOfflineRequests' from an application's datasources.json
+  // solves bug https://github.com/strongloop/loopback-next/issues/2198
+  let maxOfflineRequests = 16;
```

Comment:
We can expose `DEFAULT_MAX_OFFLINE_REQUESTS` as a property of `DataSource` class itself so that the global default can be changed as well.